blazor/datagrid/connecting-to-data-source/microsoft-sql-server.md

```diff
@@ -0,0 +1,641 @@
+    </GridColumns>
+</SfGrid>
+
+@code {
+    SfGrid<Order> Grid { get; set; }
+}
+{% endhighlight %}
+{% highlight razor tabtitle="Orderdata.cs"%}
+  public class Order
+  {
+      public int? OrderID { get; set; }
+      public string CustomerID { get; set; }
+      public int EmployeeID { get; set; }
+      public decimal Freight { get; set; }
+      public string ShipCity { get; set; }
+  }
+{% endhighlight %}
+{% endtabs %}
+
+**4.** Within the custom adaptor’s ReadAsync method, leverage the [DataManagerRequest](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.DataManagerRequest.html) to gather Grid action specifics such as paging, filtering, and sorting details.
+
+* Utilize the `DataManagerRequest` to formulate a SQL query string for paging purposes and execute the query. Fetch data from the database using the **SqlDataAdapter** class.
+
+* Employ the Fill method of the **DataAdapter** to populate a DataSet with the results of the `Select` command of the DataAdapter, followed by conversion of the DataSet into a List.
+
+* Finally, return the response as a `Result` and `Count` pair object in the ReadAsync method to bind the data to the DataGrid.
+
+{% tabs %}
+{% highlight razor tabtitle="Index.razor" %}
+@rendermode InteractiveServer
+
+@using Syncfusion.Blazor.Grids
+@using Syncfusion.Blazor.Data
+@using Syncfusion.Blazor
+@using Microsoft.Data.SqlClient;
+
+<SfGrid @ref="Grid" TValue="Order" AllowSorting="true" AllowFiltering="true"  AllowGrouping="true" AllowPaging="true" Toolbar="@(new List<string>() { "Add", "Delete", "Update", "Cancel" })">
+    <SfDataManager AdaptorInstance="@typeof(CustomAdaptor)" Adaptor="Adaptors.CustomAdaptor"></SfDataManager>
+    <GridEditSettings AllowEditing="true" AllowDeleting="true" AllowAdding="true" Mode="@EditMode.Normal"></GridEditSettings>    
+    <GridColumns>
+        <GridColumn Field=@nameof(Order.OrderID) HeaderText="Order ID" IsIdentity="true" IsPrimaryKey="true" TextAlign="TextAlign.Right" Width="120"></GridColumn>
+        <GridColumn Field=@nameof(Order.CustomerID) HeaderText="Customer Name" Width="150"></GridColumn>
+        <GridColumn Field=@nameof(Order.EmployeeID) HeaderText="Employee ID" Width="150"></GridColumn>
+        <GridColumn Field=@nameof(Order.Freight) HeaderText="Freight" Width="150"></GridColumn>       
+        <GridColumn Field=@nameof(Order.ShipCity) HeaderText="Ship City" Width="150"></GridColumn>
+    </GridColumns>
+</SfGrid>
+
+@code {
+    SfGrid<Order> Grid { get; set; }
+    
+     public class CustomAdaptor : DataAdaptor
+  {
+      public OrderData Order = new OrderData();
+      // Performs data Read operation
+      public override async Task<object> ReadAsync(DataManagerRequest dm, string key = null)
+      {
+          IEnumerable<Order> DataSource = await Order.GetOrdersAsync();    
+          int count = DataSource.Cast<Order>().Count();
+          if (dm.Skip != 0)
```

Comment:
need to provide information regarding skip and take as like sorting, filtering etc